syntax = "proto3";

package resemble.v1alpha1;

////////////////////////////////////////////////////////////////////////

message TaskId {
  string service = 1;
  string actor_id = 2;
  bytes task_uuid = 3;
}

////////////////////////////////////////////////////////////////////////

message WaitRequest {
  TaskId task_id = 1;
}

message WaitResponse {
  bytes response = 1;
}

////////////////////////////////////////////////////////////////////////

message ListPendingTasksRequest {}

message ListPendingTasksResponse {
  repeated TaskId task_ids = 1;
}

////////////////////////////////////////////////////////////////////////

service Tasks {
  rpc Wait(WaitRequest) returns (WaitResponse);
  rpc ListPendingTasks(ListPendingTasksRequest)
      returns (ListPendingTasksResponse);
}

////////////////////////////////////////////////////////////////////////
